server/db/dbManager.ts

```diff
@@ -24,15 +24,23 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(parent?: mongoose.Types.ObjectId): Promise<models.IPost[]> {
+	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
 		let posts;
 		if (parent) {
-			posts = this.model.find({ parent });
+			posts = await this.model.find({ parent }).sort({ upvotes: -1 }).lean().exec();
+			if (depth < parseInt(process.env.MAX_COMMENT_DEPTH)) {
+				for (const post of posts) {
+					post.children = [];
+					if (depth > 0) {
+						parentObject.children.push(post);
+					}
+					await this.getAllPosts(post._id, depth + 1, post);
+				}
```

Comment:
If you see `await` in the `for` you should think twice. And then 10th time if needed. 😄

If you have a lot of queries, think about how to make it recursively inside the single DB query.
Perhaps, take a look at something like this: https://docs.mongodb.com/manual/reference/operator/aggregation/graphLookup/.
Of course, it is a bit opposite example, but still recursive. You may also combine several MongoDB functions in a single request.